modules/core/src/lib/effect-manager.ts

```diff
@@ -16,9 +17,10 @@ export default class EffectManager {
     this._setEffects([]);
   }
 
-  addDefaultEffect(effect: Effect) {
-    if (!this._defaultEffects.find(e => e.constructor === effect.constructor)) {
-      this._defaultEffects.push(effect);
+  addDefaultEffect(effect: Effect, slot: number) {
```

Comment:
Are there really many people out there writing custom effects? 

My understanding is that the purpose of the `deck._addDefaultEffect` API is to be a internal way to allow extensions to register a default effect without having to bloat the core. Having a full dependency graph as well as an order seems overkill. As you mentioned elsewhere, we could rethink the Effect API in the future. As for a stopgap, how about spacing the internal effects so that an external effect can always be placed inbetween

```js
const EffectOrder = {
  MaskEffect: 10,
  CollideEffect: 20
}
```

Reply:
Ok. Can we at least make it so that effects are sorted by this number, instead of overwritten?